adapter-tcp/src/main/java/io/nosqlbench/adapter/tcpserver/TcpServerAdapterSpace.java

```diff
@@ -0,0 +1,291 @@
+
+
+        public SocketWriter(BlockingQueue<String> sourceQueue, Socket connectedSocket) {
+            this.sourceQueue = sourceQueue;
+            try {
+                outputStream = connectedSocket.getOutputStream();
+                this.writer = new OutputStreamWriter(outputStream);
+                //connectedSocket.shutdownInput();
+            } catch (Exception e) {
+                throw new RuntimeException(e);
+            }
+        }
+
+        public void shutdown() {
+            this.running = false;
+        }
+
+        @Override
+        public void run() {
+            try (Writer writer = this.writer) {
+                while (running) {
+                    while (!sourceQueue.isEmpty() ) {
+                        try {
+                            String data = sourceQueue.take();
+                            writer.write(data);
+                            writer.flush();
+                        } catch (Exception e) {
+                            throw new RuntimeException(e);
+                        }
+                    }
+                    try {
+                        Thread.sleep(10);
+                    } catch (InterruptedException ignored) {
+                    }
+                }
+            } catch (Exception e) {
+                throw new RuntimeException(e);
+            }
+            finally {
+                //TODO: Add the shutdown logic to do after closing the socket
+            }
+
+        }
+
+    }
+
+    public static class QueueWriterAdapter extends Writer {
+        private BlockingQueue<String> queue;
+
+        public QueueWriterAdapter(BlockingQueue<String> queue) {
+            this.queue = queue;
+        }
+
+        @Override
+        public synchronized void write( char[] cbuf, int off, int len) {
+            while (true) {
+                try {
+                    queue.put(new String(cbuf, off, len));
+                    return;
+                } catch (InterruptedException ignored) {
+                } catch (Exception e) {
+                    throw new RuntimeException(e);
+                }
+            }
+        }
+
+        @Override
+        public synchronized void flush() throws IOException {
+        }
+
+        @Override
+        public synchronized void close() throws IOException {
+            flush();
+            queue = null;
+        }
+
+    }
+
+    public class SocketAcceptor implements Runnable, Shutdown {
+        private final BlockingQueue<String> queue;
+        private final ServerSocket serverSocket;
+        private boolean running = true;
+
+        public SocketAcceptor(BlockingQueue<String> queue, ServerSocket serverSocket) {
+            this.queue = queue;
+            this.serverSocket = serverSocket;
+        }
+
+        public void shutdown() {
+            this.running = false;
+        }
+
+        @Override
+        public void run() {
+            try (ServerSocket serverSocket = this.serverSocket) {
```

Comment:
Let's change the name to runServerSocket or threadServerSocket to avoid confusion with the instance variable `serverSocket` defined as an instance variable.